src/accelerate/accelerator.py

```diff
@@ -1995,7 +1995,7 @@ def _prepare_ipex_or_xpu(self, *args):
                 optimizer = obj
         if optimizer is not None and model is not None:
             dtype = torch.bfloat16 if self.state.mixed_precision == "bf16" else None
-            if self.device.type == "xpu" and model.device.type == "cpu":
+            if self.device.type == "xpu" and hasattr(model, "device") and model.device.type == "cpu":
```

Comment:
Couldn't we also just check the parameters like we do normally? E.g.:

```python
if self.device.type == "xpu" and next(model.parameters()).device.type == "cpu"
```

This way if it has it or not doesn't matter.

Reply:
thanks for the suggestion! code updated and verified on XPU.